resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterManager.scala

```diff
@@ -96,16 +99,30 @@ private[spark] class KubernetesClusterManager extends ExternalClusterManager wit
         sparkConf,
         Some(new File(Config.KUBERNETES_SERVICE_ACCOUNT_TOKEN_PATH)),
         Some(new File(Config.KUBERNETES_SERVICE_ACCOUNT_CA_CRT_PATH)))
+
+    val kubernetesShuffleManager = if (Utils.isDynamicAllocationEnabled(sparkConf)) {
+      val kubernetesExternalShuffleClient = new KubernetesExternalShuffleClientImpl(
```

Comment:
Does this ensure that each executor gets a different instance of the shuffle client?

Reply:
`KubernetesClusterManager` is on the driver.